app/services/accept_proposed_organisation.rb

```diff
@@ -21,17 +23,35 @@ def initialize(proposed_org)
   end
 
   def run
-    org = @proposed_org.accept_proposal
-    usr = User.find_by(email: @email)
-    if usr
-      NotifyRegisteredUserFromProposedOrg.new(usr,org).run
-      Response.new(Response::NOTIFICATION_SENT,nil, org)
-    else
-      create_invitation_response_object(InviteUnregisteredUserFromProposedOrg.new(@email,org).run, org)
+    begin
+      accept_organization
+    rescue Exception
```

Comment:
Could we rescue a specific error here rather than `Exception`?

Also, you can remove the `begin ... rescue ... end`, and just have `rescue` to spaces to the left. http://seejohncode.com/2012/04/17/short-tip-rescuing-a-method-in-ruby/